db.go

```diff
@@ -709,16 +709,37 @@ func (db *DB) blocksForInterval(mint, maxt int64) []Block {
 	return bs
 }
 
+// openHeadBlock opens the head block at dir.
+func (db *DB) openHeadBlock(dir string) (*HeadBlock, error) {
+	var (
+		wdir = filepath.Join(dir, "wal")
+		l    = log.With(db.logger, "wal", wdir)
+	)
+	wal, err := OpenSegmentWAL(wdir, l, 5*time.Second)
+	if err != nil {
+		return nil, errors.Wrap(err, "open WAL %s")
+	}
+
+	h, err := OpenHeadBlock(dir, log.With(db.logger, "block", dir), wal)
+	if err != nil {
+		return nil, errors.Wrapf(err, "open head block %s", dir)
+	}
+	return h, nil
+}
+
 // cut starts a new head block to append to. The completed head block
 // will still be appendable for the configured grace period.
-func (db *DB) cut(mint int64) (HeadBlock, error) {
+func (db *DB) cut(mint int64) (headBlock, error) {
 	maxt := mint + int64(db.opts.MinBlockDuration)
 
 	dir, seq, err := nextSequenceFile(db.dir, "b-")
 	if err != nil {
 		return nil, err
 	}
-	newHead, err := createHeadBlock(dir, seq, db.logger, mint, maxt)
+	if err := TouchHeadBlock(dir, seq, mint, maxt); err != nil {
+		return nil, errors.Wrapf(err, "touch head block %s", dir)
+	}
+	newHead, err := db.openHeadBlock(dir)
```

Comment:
Can we not `touch` inside `openHeadBlock`? Seems logical as it a part of the opening.